tidy3d/components/mode/simulation.py

```diff
@@ -213,18 +213,6 @@ def plane_in_sim_bounds(cls, val, values):
             raise SetupError("'ModeSimulation.plane' must intersect 'ModeSimulation.geometry.")
         return val
 
-    @pd.validator("boundary_spec", always=True)
-    def boundaries_for_zero_dims(cls, val, values):
-        """Replace with periodic boundary along zero-size dimensions."""
-        boundaries = [val.x, val.y, val.z]
-        size = values.get("size")
-
-        for dim, size_dim in enumerate(size):
-            if size_dim == 0:
-                boundaries[dim] = Boundary.periodic()
-
-        return BoundarySpec(x=boundaries[0], y=boundaries[1], z=boundaries[2])
```

Comment:
Ah so this validator actually existed but it wasn't taking effect because `_boundaries_for_zero_dims = validate_boundaries_for_zero_dims()` was also included (and probably executed first?)